accounts/main.mo

```diff
@@ -0,0 +1,207 @@
+                        #ok({
+                            message = "linked principals to account";
+                            account = newAccount;
+                        });
+                    };
+                };
+            };
+        };
+    };
+    
+    public query({caller}) func getAccount(
+        accountId : MAccount.AccountId,
+    ) : async Result.Result<MAccount.Account, ()> {
+        switch (users.accounts.get(accountId)) {
+            case (null) { #err(); };
+            case (? a)  { #ok(a); };
+        };
+    };
+
+    public query({caller}) func getAccountDetails(
+        accountId : MAccount.AccountId,
+    ) : async Result.Result<MAccount.AccountDetails, ()> {
+        switch (users.accounts.get(accountId)) {
+            case (null) { #err(); };
+            case (? a)  { #ok(MAccount.getDetails(a)); };
+        };
+    };
+
+    public query({caller}) func getAccountsFromScores(
+        scores : [MPublic.Score],
+    ) : async [MAccount.Score] {
+        let accounts = Array.init<(MAccount.AccountId, Nat)>(scores.size(), (0 , 0));
+        for (i in scores.keys()) {
+            let (playerId, score) = scores[i];
+            switch (users.getAccountByPrincipal(MPlayer.unpack(playerId))) {
+                case (null)      {};
+                case (? account) {
+                    accounts[i] := (account.id, score);
+                };
+            };
+        };
+        Array.freeze(accounts);
+    };
+    
+    public shared({caller}) func updateAccount(
+        req : MAccount.UpdateRequest,
+    ) : async MAccount.UpdateResponse {
+        switch (users.getAccountByPrincipal(caller)) {
+            case (null)      {
+                #err("account not found: " # Principal.toText(caller));
+            };
+            case (? account) {
+                let updatedAccount : MAccount.Account = {
+                    alias         = req.alias;
+                    avatar        = req.avatar;
+                    flavorText    = req.flavorText;
+                    id            = account.id;
+                    plugAddress   = account.plugAddress;
+                    primaryWallet = switch (req.primaryWallet) {
+                        case (null) { account.primaryWallet; };
+                        case (? newWallet) { newWallet; };
+                    };
+                    stoicAddress  = account.stoicAddress;
+                };
+                users.putAccount(updatedAccount);
+                #ok(updatedAccount);
+            };
+        };
+    };
+
+    // ◤━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━◥
+    // | Admin zone. 🚫                                                        |
+    // ◣━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━◢
+
+    private stable var stableAdmins : [(Principal, [Roles.Role])] = [(owner, [Roles.ALL])];
+    private let admins = Roles.Users(stableAdmins);
```

Comment:
The auth package seems neat, but do we need it for something at this time? I think we only have the one role, which wasn't really used during the last tournament. I'm not against it if it doesn't add complexity.